listenbrainz/mbid_mapping_writer/mbid_mapper.py

```diff
@@ -221,65 +312,34 @@ def search(self, artist_credit_name, recording_name):
-            hit = self.lookup(artist_credit_name_p, recording_name_p)
+        # lookup with only artist credit detuned
+        if ac_detuned:
+            self._log("Detune only artist_credit")
+            hit = self.lookup_and_evaluate_hit(ac_detuned, recording_name_p, True, False)
             if hit:
-                (hit, match_type) = self.evaluate_hit(
-                    hit, artist_credit_name_p, recording_name_p)
-                if hit:
-                    self._log(Markup(f"""
-                        <table>
-                            <tr>
-                                <th>recording name</th>
-                                <th>release name</th>
-                                <th>artist credit name</th>
-                                <th>recording MBID</th>
-                                <th>release MBID</th>
-                                <th>artist_credit_id</th>
-                            </tr>
-                            <tr>
-                                <td>{hit['document']["recording_name"]}</td>
-                                <td>{hit['document']["release_name"]}</td>
-                                <td>{hit['document']["artist_credit_name"]}</td>
-                                <td>{hit['document']["recording_mbid"]}</td>
-                                <td>{hit['document']["release_mbid"]}</td>
-                                <td>{hit['document']["artist_credit_id"]}</td>
-                            </tr>
-                        </table>
-                    """))
-
-            if not hit:
-                hit = None
-                if ac_detuned:
-                    artist_credit_name_p = ac_detuned
-                    self.is_ac_detuned = True
-                    self._log("Detune artist_credit")
-                    ac_detuned = None
-                    continue
-
-                if r_detuned:
-                    recording_name_p = r_detuned
-                    self.is_r_detuned = True
-                    r_detuned = None
-                    self._log("Detune recording")
-                    continue
-
-                self._log("FAIL (if this is the only line of output, it means we literally have no clue what this is)")
+                return hit
 
-                return None
+        # lookup with both artist credit and recording detuned
+        if ac_detuned and r_detuned:
+            self._log("Detune artist_credit and recording")
+            hit = self.lookup_and_evaluate_hit(ac_detuned, r_detuned, True, True)
+            if hit:
+                return hit
 
-            break
+        # this case is the last one because it didn't exist in earlier versions and
+        # preserving order of cases with older versions is probably sensible.
+        if r_detuned:
+            self._log("Detune only recording")
+            hit = self.lookup_and_evaluate_hit(artist_credit_name_p, r_detuned, False, True)
+            if hit:
+                return hit
```

Comment:
This is the new case. Everything else in the PR is aimed to be a refactoring preserving existing behaviour.